### PR TITLE
remove use of deprecated np.int

### PR DESCRIPTION
**Pull request by javierggt**
## Description

This PR makes tiny changes to remove uses of a few deprecated dtypes (https://github.com/sot/skare3/issues/753)

## Testing

- [x] Passes unit tests on MacOS, linux, Windows (at least one required)
- [ ] Functional testing

Fixes #

### CODE REVIEW

**Comment by taldcroft** [comment]
I thought the recommended change is `np.int` => `int`. From the [release notes](https://numpy.org/devdocs/release/1.20.0-notes.html#using-the-aliases-of-builtin-types-like-np-int-is-deprecated), "For a long time, `np.int` has been an alias of the builtin `int`".

**Comment by taldcroft** [comment]
And just to see for myself:
```
In [1]: np.int is int                                                           
Out[1]: True

In [2]: np.float is float                                                       
Out[2]: True

In [3]: isinstance(np.float64(10), float)                                       
Out[3]: True
```

**Comment by javierggt** [comment]
I would agree that perhaps `np.int_` is not the right type. I changed it to `int`.

In other places (Annie, I believe) I wondered whether it should be `np.int32` to be explicit. `np.int32` is used in Annie in several places.

Also, the case is clear for `bool`, but if one uses `int`, the type might be platform dependent.

File: Ska/Numpy/Numpy.py [hunk 1]
```diff
@@ -71,7 +71,7 @@ def match(recarray, filters):
     except TypeError:
         pass
 
-    matches = np.ones(len(recarray), dtype=np.bool)
+    matches = np.ones(len(recarray), dtype=np.bool_)
```
Comment (taldcroft) [inline Ska/Numpy/Numpy.py]: Should be `bool`.

**Review by taldcroft — APPROVED** [review]

**Pull request merged**